### PR TITLE
Add support for magna and all dial funcs

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: internal/binding/local.go [hunk 1]
```diff
@@ -188,20 +437,44 @@ func (p *chanPort) ReadPacketData() ([]byte, gopacket.CaptureInfo, error) {
...
+// dutPortMgr implements lemming's port creator interface.
+// Note: this interface is different from magna, to avoid making the two project depending on eachother.
+type dutPortMgr struct {
+	ports map[string]*chanPort
+	// TODO: lastCreatedPort is a hack because we don't currently model hardware ports correctly in lemming.
 	lastCreatedPort *chanPort
 }
 
-func (lm *portMgr) CreatePort(name string) (fwdcontext.Port, error) {
+func (lm *dutPortMgr) CreatePort(name string) (fwdcontext.Port, error) {
```
Comment (wenovus) [inline internal/binding/local.go:477]: Should we use a different alias than `lm`?

File: internal/binding/local.go [hunk 2]
```diff
@@ -77,80 +218,188 @@ func (lb *LocalBind) Reserve(ctx context.Context, tb *opb.Testbed, _, _ time.Dur
...
+	}
+
+	otgSrv.AddFlowHandlers(fh)
+	otgSrv.AddFlowHandlers(ipFH)
+	telemSrv.AddTask(task)
+	telemSrv.AddTask(ipTask)
+
+	hintCh := make(chan lwotg.Hint, 100)
+	otgSrv.SetHintChannel(hintCh)
+	// otgSrv.SetProtocolHandler(gatewayPinger)
```
Comment (wenovus) [inline internal/binding/local.go:351]: debug?

**Review by wenovus — APPROVED** [review]

File: internal/binding/local.go (hunk 1 above)
Comment (DanG100) [reply internal/binding/local.go]: changed to mgr

File: internal/binding/local.go (hunk 2 above)
Comment (DanG100) [reply internal/binding/local.go]: removed

**Pull request merged**